smartapps/ady624/webcore.src/webcore.groovy

```diff
@@ -813,69 +578,88 @@ def installed() {
+		registerInstance()
+	}
+	def recoveryMethod = (settings.recovery ?: 'Every 30 minutes').replace('Every ', 'Every').replace(' minute', 'Minute').replace(' hour', 'Hour')
+	if (recoveryMethod != 'Never') {
+		try {
+			"run$recoveryMethod"(recoveryHandler)
+		} catch (all) { }
+	}
 
-    //move lifx
-    if (state.settings && state.settings.lifx_scenes) {
-    	state.lifx = [
-        	scenes: state.settings.lifx_scenes,
-            lights: state.settings.lifx_lights,
-            groups: state.settings.lifx_groups,
-            locations: state.settings.lifx_locations
-        ]
-        state.settings.remove('lifx_scenes')
-        state.settings.remove('lifx_lights')
+	//move lifx
+	if (state.settings && state.settings.lifx_scenes) {
+		state.lifx = [
+			scenes: state.settings.lifx_scenes,
+			lights: state.settings.lifx_lights,
+			groups: state.settings.lifx_groups,
+			locations: state.settings.lifx_locations
+		]
+		state.settings.remove('lifx_scenes')
+		state.settings.remove('lifx_lights')
 		state.settings.remove('lifx_groups')
 		state.settings.remove('lifx_locations')
 	}
-    
-    if(state.accessToken){
-    	updateEndpoint(state.accessToken)
-    }
+	
+	if(state.accessToken){
+		updateEndpoint(state.accessToken)
+	}
 }
 
 private updateEndpoint(accessToken){
-    if(isCustomEndpoint()){
-        state.endpoint = customServerUrl("?access_token=${accessToken}")
-    }
-    else {
-        state.endpoint = isHubitat() ? apiServerUrl("$hubUID/apps/${app.id}/?access_token=${accessToken}") : apiServerUrl("/api/token/${accessToken}/smartapps/installations/${app.id}/")
-    }
+	if(isCustomEndpoint()){
+		state.endpoint = customServerUrl("?access_token=${accessToken}")
+	}
+	else {
+		state.endpoint = isHubitat() ? apiServerUrl("$hubUID/apps/${app.id}/?access_token=${accessToken}") : apiServerUrl("/api/token/${accessToken}/smartapps/installations/${app.id}/")
+	}
 }
+
 private initializeWebCoREEndpoint() {
 	try {
-        if (!state.endpoint) {
-            try {
-                def accessToken = createAccessToken()
-                if (accessToken) {
-                    updateEndpoint(accessToken)
-                }
-            } catch(e) {
-                state.endpoint = null
-            }
-        }
-        return state.endpoint
+		if (!state.endpoint) {
+			try {
+				def accessToken = createAccessToken()
+				if (accessToken) {
+					updateEndpoint(accessToken)
+				}
+				else {
+					enableOauth()
+					return initializeWebCoREEndpoint()
+				}
+			} catch(e) {
+				state.endpoint = null
+			}
+		}
+		return state.endpoint
 	} catch (all) {
-    	error "An error has occurred during endpoint initialization: ", all
-    }
-    return false
+		error "An error has occurred during endpoint initialization: ", all
+	}
+	return false
+}
+
+private enableOauth() {
+	def params = [
+		uri: "http://localhost:8080/app/edit/update?_action_update=Update&oauthEnabled=true&id=${app.appTypeId}",
```

Comment:
This is a security violation. The app should never update it's own settings without the user doing it themselves.

Reply:
We should chat in some less public forum on where I found that.  In this case, webcore will not work without it, so it is mandatory to use it.

Reply:
Right, we can alert the user to enable it, but we shouldn't emulate a browser to do it without the user's permission.

Reply:
I think the permission is already there in all the notifications on web enabled app on install.

Reply:
The user gives consent, but this is uses a backdoor/unsupported call to accomplish it.

Reply:
We can ask a question about to enable, but that appears to be redundant in flow since they are agreeing so we ask them again in they agree....A lot of apps use this